examples/showcase/tracked-controls/js/components/vive-controls.js

```diff
@@ -0,0 +1,117 @@
+    var self = this;
+    this.animationActive = 'pointing';
+    this.onButtonChanged = this.onButtonChanged.bind(this);
+    this.onButtonDown = function (evt) { self.onButtonEvent(evt.detail.id, 'down'); };
+    this.onButtonUp = function (evt) { self.onButtonEvent(evt.detail.id, 'up'); };
+    this.onModelLoaded = this.onModelLoaded.bind(this);
+  },
+
+  play: function () {
+    var el = this.el;
+    el.addEventListener('buttonchanged', this.onButtonChanged);
+    el.addEventListener('buttondown', this.onButtonDown);
+    el.addEventListener('buttonup', this.onButtonUp);
+    el.addEventListener('axisMove', this.onButtonUp);
+    el.addEventListener('model-loaded', this.onModelLoaded);
+  },
+
+  pause: function () {
+    var el = this.el;
+    el.removeEventListener('buttonchanged', this.onButtonChanged);
+    el.removeEventListener('buttondown', this.onButtonDown);
+    el.removeEventListener('buttonup', this.onButtonUp);
+    el.removeEventListener('axisMove', this.onButtonUp);
+    el.removeEventLister('model-loaded', this.onModelLoaded);
+  },
+
+  update: function () {
+    var el = this.el;
+    var data = this.data;
+    var objUrl = 'url(https://media.aframe.io/controllers/vive/vr_controller_vive.obj)';
+    var mtlUrl = 'url(https://media.aframe.io/controllers/vive/vr_controller_vive.mtl)';
+    // handId: 0 - right, 1 - left
+    var controller = data.hand === 'right' ? 0 : 1;
+    el.setAttribute('tracked-controls', 'controller', controller);
+    if (!data.model) { return; }
+    el.setAttribute('obj-model', {obj: objUrl, mtl: mtlUrl});
+  },
+
+  onButtonChanged: function (evt) {
+    var button = this.mapping['button' + evt.detail.id];
+    var buttonMeshes = this.buttonMeshes;
+    var value;
+    if (button !== 'trigger' || !buttonMeshes) { return; }
+    value = evt.detail.state.value;
+    buttonMeshes.trigger.rotation.x = -value * (Math.PI / 8);
+  },
+
+  onModelLoaded: function (evt) {
+    var controllerObject3D = evt.detail.model;
+    var buttonMeshes;
+    if (!this.data.model) { return; }
+    buttonMeshes = this.buttonMeshes = {};
+    buttonMeshes.grip = {
+      left: controllerObject3D.getObjectByName('leftgrip'),
+      right: controllerObject3D.getObjectByName('rightgrip')
+    };
+    buttonMeshes.menu = controllerObject3D.getObjectByName('menubutton');
+    buttonMeshes.system = controllerObject3D.getObjectByName('systembutton');
+    buttonMeshes.trackpad = controllerObject3D.getObjectByName('touchpad');
+    buttonMeshes.trigger = controllerObject3D.getObjectByName('trigger');
+    // Offset pivot point
+    controllerObject3D.position.set(0, -0.04, -0.015);
+  },
+
+  onButtonEvent: function (id, evtName) {
+    var buttonName = this.mapping['button' + id];
+    this.el.emit(buttonName + evtName);
+    if (!this.data.model) { return; }
+    this.updateModel(buttonName, evtName);
+  },
+
+  updateModel: function (buttonName, state) {
+    var color = state === 'up' ? 'white' : 'yellow';
```

Comment:
At least configurable. Off white is a bit easier on the eyes, and yellow is a bit too contrasty.